### PR TITLE
Nerfs Blacksmith Imbuing because I got paxxed and lost a fight.

### PR DESCRIPTION
**Pull request by shayoki**
## About The Pull Request

Reverts some of this: https://github.com/Skyrat-SS13/Skyrat-tg/pull/25861

This PR reintroduces the Disallowed Reagents. A blacklist that was previously removed to make it possible to imbue all Blacksmithing forged gear with any reagent; is back. The PR mentioned above also frees every materials type to be used. This PR does not restrict material types. 

You can still make pizza boots imbued with soy sauce. 

## Why It's Good For The Game

I have never heard from anyone affected by pax weapons say it was a fun and fair fight.

As it stands right now, the meta to deal with threatening antagonists or security officers, is to bap them once with any reagent weapon with pax imbued, to instantly pacify them and win fights. A single attack of a fully 250 unit imbued weapon will instantly inject 1 unit of whichever reagent. And pax is a slow reagent to process.

Additionally, this PR adds more chems to the original blacklist, such the aphrodisiac list, which covers crocin and more that should not be wildly injected into players, according to their preferences.
<details>
<summary>Full list of banned reagents and reasons for being blacklisted</summary>
<br>- All inverse reagents. Some inverse chems like Super Melatonin, Corazargh, and Cryogelidia are too powerful when used in combat.
<br>- Entropic Polypnium. A powerful hallucinogen that deals oxygen, stamina, and visual effects per tick.
<br>- Pax. This is a controversial chem that is used to pacify anyone who can accept reagents. This is a slow-to-process chem and having no option to attack back leaves the defender too vulnerable. Once paxxed your only option is to run away or attempt to disable the attacker non-lethally. 
<br>- Liquid Electricity, Teslium, Yuck, Spewium, Rat spit. These reagents when used in combat can be used to stun-lock the opponent. The first two stuns the character through electricity shocks. And the last two cause immediate nausea that cause a character to puke and stand still for about ten seconds.
<br>-Eigenstate. This is used to near instantly teleport an opponent to either a random location or a set location. Depending on where the chemical were brewed, This can send anyone attacked with an imbued weapon to a trap an opponent.
<br>- Bodily fluids and Aphrodisiacs. Obviously we don't want to inject others with lewd drugs against their will as a prefbreak.
<br>- Polonium. 5 units in anyone instantly makes someone radioactive, the affects stay until toxin treatment is administered, hair will burn off, and the flesh of the ones affected will burn too.
<br>- Mutagen. This gives a random DNA mutation when injected. Mutadone must be administered as a treatment.
<br>- Zauker. This chem deals 50 damage across brute, burn, and toxins in a single unit. This reagent should be impossible to obtain but this is here in case.
<br>- PCP. The only stimulant being blacklisted due to its many plethora of benefits as a combat stim and benefits against those not expecting it.

</details>

## Proof Of Testing

It just works.
fairandbalanced

## Changelog

:cl:
add: Reverts the Forging Imbuing reagent blacklist removal.
/:cl:

### CODE REVIEW

**Review by projectkepler-ru — APPROVED** [review]

**Review by Majkl-J — COMMENTED** [review]
Are we sure this list needs to be brought back in full?

**Comment by Majkl-J** [comment]
> Poisons such as mute toxin, lexorin, fentanyl and most found in the syndicate poison kit are still fine to use and imbue.
> 
> I am considering taking out earthsblood and other stimulants from the list.

Feel free to make adjustments to the list accordingly

**Comment by YakumoChen** [comment]
> Additionally, this PR adds more chems to the original blacklist, such the aphrodisiac list, which covers crocin and more that should not be wildly injected into players, according to their preferences.

Don't agree this needs to happen, making prefbreak weapons already risks a permanent ban if misused, but that doesn't mean we need to axe legitimate uses.

**Comment by PawnAtWork** [comment]
> Don't agree this needs to happen, making prefbreak weapons already risks a permanent ban if misused, but that doesn't mean we need to axe legitimate uses.
> 
> It's incredibly easy to catch as well since this would be in the attack log (both injections and weapon hits)

100% Agree. Why punish people because they planned their attack out? We're just going to lose people who want to play antag. It's already pain enough that people are harder to kill than standard /TG/ we don't need to make it harder, pax is designed to fuck you over... It's not meant to be a fair fight. If you get paxxed, RUN. If you can't run, turn your sensors on and pray medical is paying attention.

**Comment by ReturnToZender** [comment]
This is great, thank you Gavla

**Comment by NargaInDisguise** [comment]
THANK FUCKING GOD someones finally making this nerf PR, forging unleashed was such a terrible pr balance-wise.
Putting pax and synth-pax on weapons has not once ever been considered "fun" for anyone.

This PR (https://github.com/Skyrat-SS13/Skyrat-tg/pull/25861) had good ideas, like letting you make weapons outta wacky shit, but removing the entire chem blacklist was hilariously bad.

**Comment by NargaInDisguise** [comment]
Also the infinite damage blacksmithing bug is STILL present and being abused, apparently.
May need to check on that.

**Comment by Cyprex** [comment]
This feature already lets you make weapons that do a gazillion damage- reagents make it even worse. Honestly, just make this a whitelist instead.

**Comment by Cyprex** [comment]
I am going to say that nothing made via smithing should be stronger than a fireaxe, or anything a skilled botanist can make in 30 minutes without using strange seeds.

**Comment by projectkepler-ru** [comment]
> This feature already lets you make weapons that do a gazillion damage- reagents make it even worse. Honestly, just make this a whitelist instead.

zirok that's a bug you're not supposed to be able to infinitely stack damage

**Review by projectkepler-ru — COMMENTED** [review]
In the future do refrain from making PR Title as egregious as this

**Comment by NargaInDisguise** [comment]
> In the future do refrain from making PR Title as egregious as this

This is like, a long-standing tradition with bubber nerf prs at this point, burger started a trend a while back.

**Comment by projectkepler-ru** [comment]
> This is like, a long-standing tradition with bubber nerf prs at this point, burger started a trend a while back.

I can largely forgive burger as they're not the one losing a fight using sleeping carp

File: modular_skyrat/modules/reagent_forging/code/forge.dm [hunk 1]
```diff
@@ -91,6 +91,30 @@
...
+		/datum/reagent/yuck,
+		/datum/reagent/toxin/spewium,
+		/datum/reagent/eigenstate,
+		/datum/reagent/drug/pcp,
+		/datum/reagent/consumable/cum,
+		/datum/reagent/consumable/femcum,
+		/datum/reagent/consumable/breast_milk,
+		/datum/reagent/drug/aphrodisiac, 			//This path covers all lewd chems such as crocin, hexacrocin, succubus milk, and more.
+		/datum/reagent/toxin/acid,
+		/datum/reagent/clf3,
```
Comment (Arturlang) [inline modular_skyrat/modules/reagent_forging/code/forge.dm:111]: why acid, clf3, phlog, napalm? Do they deal too much damage?

**Comment by Arturlang** [comment]
Put a list of all your disallow reagents on the PR body so people don't have to codedive what you're blacklisting

**Pull request closed without merging**